Disable Rapid Shot on Throwing weapons and adjust delay reduction

Flam9 opened the pull request:
Sources: 
- https://www.bg-wiki.com/ffxi/Rapid_Shot
- https://wiki-ffo-jp.translate.goog/html/2986.html?_x_tr_sl=ja&_x_tr_tl=en&_x_tr_hl=en
- https://www.ffxiah.com/forum/topic/49806/ranger-firing-range-testing/4/#3233650

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Disable Rapid Shot on Throwing Weapons
- Adjust delay to 2-50% from 6-50%

## Steps to test these changes

Gear
- Bandits gun lvl 15: `!giveitem Flam 17257 1`
- Bullet: `!giveitem Flam 17340 99`
- Dart: `!giveitem Flam 17307 99`

Steps
- Be a RNG or COR (`!changejob 17`) lvl 15+
- Equip darts, spam `/ra <t>` and time attack durations with stopwatch. Notice no changes after ~20 autos.
- Equip gun, do ^

WinterSolstice8 commented:
Ugh, you're telling me BG has two pages that say two different things and one matches JP wiki and that's the _wrong_ reduction for rapid shot? nice...

At least Byrth had a better explanation that actually makes sense

WinterSolstice8 commented on src/map/ai/states/range_state.cpp:
```diff
@@ -72,26 +72,32 @@ CRangeState::CRangeState(CBattleEntity* PEntity, uint16 targid)
+                {
+                    // reduce delay by 2-50%
+                    // https://wiki.ffo.jp/html/2986.html
+                    // https://www.bg-wiki.com/ffxi/Delay#Ranged_Delay
```
I would remove these 2 sources and replace it with the ffxiah thread & the rapid shot bg page. Those better elucidate it

Flam9 committed:
Rapid shot trait fixes

- Disable Rapid Shot on Throwing Weapons
- Adjust delay to 2-50% from 6-50%

Sources: https://www.bg-wiki.com/ffxi/Rapid_Shot, https://wiki-ffo-jp.translate.goog/html/2986.html?_x_tr_sl=ja&_x_tr_tl=en&_x_tr_hl=en, https://www.ffxiah.com/forum/topic/49806/ranger-firing-range-testing/4/#3233650
